addons/misra.py

```diff
@@ -316,7 +333,7 @@ def hasSideEffectsRecursive(expr):
             e = e.astOperand1
         if e and e.str == '.':
             return False
-    if expr.str in ('++', '--', '='):
+    if expr.isAssignmentOp or expr.str in {'++', '--'}:
```

Comment:
do you have some reason to replace `()` with `{}`. I am not python expert.

Reply:
`{ }` is a `set`. In a set every element is unique. IMO this makes sense in this case.